client/src/components/Ticketing/ticketingmanager/AdminPurchase/AdminPurchase.tsx

```diff
@@ -0,0 +1,463 @@
+    setEventData(updatedRows);
+    // Also update the priceByRowId to reflect the change in the price
+    if (isChecked) {
+      setPriceByRowId((prevState) => ({
+        ...prevState,
+        [row.id]: '0.00', // Set to $0 if complementary
+      }));
+    }
+  };
+
+  const handlePurchase = () => {
+    // Redirect to checkout page
+  };
+
+  const getEventData = async (event) => {
+    try {
+      const token = await getAccessTokenSilently({
+        audience: process.env.REACT_APP_ROOT_URL,
+        scope: 'admin',
+      });
+      const response = await fetch(
+        process.env.REACT_APP_API_1_URL + `/doorlist?eventinstanceid=${event}`,
+        {
+          headers: {
+            'Content-Type': 'application/json',
+            'Authorization': `Bearer ${token}`,
+          },
+        },
+      );
+      const jsonRes = await response.json();
+      const jsonData = jsonRes.data;
+      const eventData = jsonData
+        .map((item, index) => {
+          const row = item.row.slice(1, -1).split(',');
+          if (!row || !row[0]) {
+            // Check if the value in column 0 is present
+            setTicketsSold(false);
+            return null; // Exit early if no tickets sold
+          }
+          setTicketsSold(true);
+          return {
+            id: index,
+            firstname: row[1],
+            lastname: row[2],
+            num_tickets: row[12],
+            arrived: false,
+            vip: row[3] === 't',
+            donorbadge: row[4] === 't',
+            accommodations: row[5],
+          };
+        })
+        .filter(Boolean);
+
+      setEventData(eventData);
+    } catch (error) {
+      console.error(error.message);
+    }
+  };
+
+  return (
+    <div className='w-full h-screen overflow-x-hidden absolute '>
+      <div className='md:ml-[18rem] md:mt-40 sm:mt-[11rem] sm:ml-[5rem] sm:mr-[5rem] sm:mb-[11rem]'>
+        <div className='flex flex-row'>
+          <h1 className='font-bold text-5xl bg-clip-text text-transparent bg-gradient-to-r from-green-500 to-zinc-500 mb-14'>
+            Purchase Tickets
+          </h1>
+        </div>
+        <div className='bg-white p-5 rounded-xl mt-2 shadow-xl'>
+          {ticketsSold ? (
+            <DataGrid
+              className='bg-white'
+              autoHeight
+              disableSelectionOnClick
+              rows={eventData}
+              columns={columns}
+              pageSize={100}
+              hideFooter
+            />
+          ) : (
+            <div className='text-xl font-bold text-red-600'>
+              No tickets sold for this show
+            </div>
+          )}
+          <div className='mt-4'>
+            <Button variant='contained' color='primary' onClick={addNewRow}>
+              Add Ticket
+            </Button>
+          </div>
+          <div className='mt-4 text-center'>
+            <Link to='/ticketing/admincheckout'>
+              <Button
+                variant='contained'
+                style={{
```

Comment:
Can you convert these to classes instead

Reply:
If its possible not to use the style attr

Reply:
Changing these to Tailwind classes causing unexpected issued. I think the MUI styles are overriding Tailwind. Any suggestions?